clients/ui/bff/internal/api/middleware.go

```diff
@@ -83,11 +84,16 @@ func resolveBearerToken(k8s integrations.KubernetesClientInterface, header http.
 	return bearerToken, nil
 }
 
-func resolveModelRegistryURL(id string, client integrations.KubernetesClientInterface) (string, error) {
-	serviceDetails, err := client.GetServiceDetailsByName(id)
+func resolveModelRegistryURL(id string, app *App) (string, error) {
+	serviceDetails, err := app.kubernetesClient.GetServiceDetailsByName(id)
 	if err != nil {
 		return "", err
 	}
+
+	if app.config.DevMode {
+		serviceDetails.ClusterIP = "localhost"
```

Comment:
Just to bring the internal conversation here, it would be really interesting to parametrize the port too, that way we can get whatever port users might have used for port-forwarding.